src/editors/_shared/hooks/DataTypes/AbstractDataType.js

```diff
@@ -118,13 +118,21 @@ class AbstractDataType {
    */
   stringEditComponent(props, placeholder, parsedValue) {
     const value = parsedValue !== undefined ? parsedValue : props.rowData.value;
+    const convert = typeof value === "object" || Array.isArray(value);
     return (
       <TextField
         inputProps={{ "data-testid": "input_value" }}
         fullWidth
         placeholder={placeholder}
-        value={value || ""}
-        onChange={evt => props.onChange(evt.target.value)}
+        defaultValue={(convert ? JSON.stringify(value) : value) || ""}
+        onChange={evt => {
+          if (!convert)
+            return props.onChange(evt.target.value);
+
+          try {
+            props.onChange(JSON.parse(evt.target.value));
+          } catch (e) {}
```

Comment:
Maybe we could move the above onChange call to it. You know, this still needs a bit of work, since I could make this work without the issues, but its still not ideal in the way that it works. Because it defaults to string type values, instead of the original type..

I'll take this into account when finishing it up

Reply:
That's resolved now